File: qc-storage/src/main/java/com/eager/questioncloud/storage/user/UserAccountInformationEntity.java
package com.eager.questioncloud.storage.user;

import com.eager.questioncloud.core.domain.user.vo.AccountType;
import com.eager.questioncloud.core.domain.user.vo.UserAccountInformation;
import jakarta.persistence.Embeddable;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Embeddable
@AllArgsConstructor
@NoArgsConstructor
public class UserAccountInformationEntity {
    private String password;
    
    private String socialUid;

    @Enumerated(EnumType.STRING)
    private AccountType accountType;

    public static UserAccountInformationEntity from(UserAccountInformation userAccountInformation) {
        return new UserAccountInformationEntity(
            userAccountInformation.getPassword(),
            userAccountInformation.getSocialUid(),
            userAccountInformation.getAccountType());
    }

    public UserAccountInformation toModel() {
        return new UserAccountInformation(password, socialUid, accountType);
    }
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/user/infrastructure/entity/UserAccountInformationEntity.kt
package com.eager.questioncloud.core.domain.user.infrastructure.entity

import com.eager.questioncloud.core.domain.user.enums.AccountType
import com.eager.questioncloud.core.domain.user.model.UserAccountInformation
import jakarta.persistence.Embeddable
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated

@Embeddable
class UserAccountInformationEntity private constructor(
    private var password: String? = null,
    private var socialUid: String? = null,
    @Enumerated(EnumType.STRING)
    private var accountType: AccountType
) {
    fun toModel(): UserAccountInformation {
        return UserAccountInformation(password, socialUid, accountType)
    }

    companion object {
        @JvmStatic
        fun from(userAccountInformation: UserAccountInformation): UserAccountInformationEntity {
            return UserAccountInformationEntity(
                userAccountInformation.password,
                userAccountInformation.socialUid,
                userAccountInformation.accountType
            )
        }
    }
}
